Mathlib/Analysis/Distribution/SchwartzSpace.lean

```diff
@@ -667,13 +667,76 @@ open MeasureTheory FiniteDimensional
 class _root_.MeasureTheory.Measure.HasTemperateGrowth (μ : Measure D) : Prop :=
   exists_integrable : ∃ (n : ℕ), Integrable (fun x ↦ (1 + ‖x‖) ^ (- (n : ℝ))) μ
 
+open Classical in
+/-- An integer exponent `l` such that `(1 + ‖x‖) ^ (-l)` is integrable if `μ` has
+temperate growth. -/
+def _root_.MeasureTheory.Measure.integrablePower (μ : Measure D) : ℕ :=
+  if h : μ.HasTemperateGrowth then h.exists_integrable.choose else 0
+
+lemma integrable_pow_neg_integrablePower
+    (μ : Measure D) [h : μ.HasTemperateGrowth] :
+    Integrable (fun x ↦ (1 + ‖x‖) ^ (- (μ.integrablePower : ℝ))) μ := by
+  simp [Measure.integrablePower, h]
+  exact h.exists_integrable.choose_spec
+
 instance _root_.MeasureTheory.Measure.IsFiniteMeasure.instHasTemperateGrowth {μ : Measure D}
     [h : IsFiniteMeasure μ] : μ.HasTemperateGrowth := ⟨⟨0, by simp⟩⟩
 
 instance _root_.MeasureTheory.Measure.IsAddHaarMeasure.instHasTemperateGrowth {μ : Measure D}
     [h : μ.IsAddHaarMeasure] : μ.HasTemperateGrowth :=
   ⟨⟨finrank ℝ D + 1, by apply integrable_one_add_norm; norm_num⟩⟩
 
+lemma pow_mul_le_of_le_of_pow_mul_le {C₁ C₂ : ℝ} {k l : ℕ} {x f : ℝ} (hx : 0 ≤ x) (hf : 0 ≤ f)
+    (h₁ : f ≤ C₁) (h₂ : x ^ (k + l) * f ≤ C₂) :
+    x ^ k * f ≤ 2 ^ l * (C₁ + C₂) * (1 + x) ^ (- (l : ℝ)) := by
```

Comment:
A final thought is that this would yield a fairly neat statement if expressed in the language of bounded functions since all the constants could be dropped.

Reply:
Not sure this would make things easier to manage, as one would need to coerce the relevant functions to spaces of bounded functions.